packages/compass-aggregations/src/modules/index.ts

```diff
@@ -319,56 +319,42 @@ const doReset = (state: RootState) => ({
  * @returns {Object} The new state.
  */
 const doRestorePipeline = (state: RootState, action: AnyAction): RootState => {
-  const savedState = action.restoreState;
-  const commenting =
-    savedState.comments === null || savedState.comments === undefined
-      ? true
-      : savedState.comments;
-  const sampling =
-    savedState.sample === null || savedState.sample === undefined
-      ? true
-      : savedState.sample;
-  const autoPreviewing =
-    savedState.autoPreview === null || savedState.autoPreview === undefined
-      ? true
-      : savedState.autoPreview;
+  const {
+    id,
+    name,
+    comments,
+    sample,
+    autoPreview,
+    collation,
```

Comment:
Just a random thought, not suggesting to do anything about it (unless is trivial), but do we need to save both `collation` and `collationString` seems one of those situations where you could derive one from another, especially for load/save, and make sure they never go out of sync by removing the derived one from the state.

Reply:
Good point! I'll take a closer look if we can improve this easily

Reply:
I moved all this collation state to one slice so that at least it's all synchronized through one action and is not two independent store slices with their own actions that can go out of sync.

Not super happy that I couldn't keep only the string collation in the store, but idiomatically we would want the derived value to come from a memoized selector so we don't need to do string parsing all the time, but compass doesn't use any and I don't want to introduce yet another pattern to the current stores that we are using, so doing this in reducer is the next best thing here I think. And we do solve the issue of storing duplicated values for saved aggregations, without breaking compat with existing items.

Did it in https://github.com/mongodb-js/compass/pull/3186/commits/b015d445ca9d7200c3ecfe85682e94a24e2c6b69 if you have a moment take a look and tell me what you think (I can revert if this doesn't feel like a better way of doing this)

Reply:
Seems an improvement to unify the action for text / value.

Just throwing ideas .. we could also only keep the object only for collation and just say that is invalid in the text input and never emit a new value if is not well-formed, more or less what we do in the connection form with the connection string.

Parsing it all the time or run / preview shouldn't be the end of the world either, is a small doc not sure is worth profiling it but should be negligible compared to the rest. There should be no need to optimize that, unless we parse on render

Reply:
Yeah what we did is already much better!